eos_data_distribution/ndn/file.py

```diff
@@ -274,6 +264,44 @@ def _on_complete(self):
         os.unlink(self._sgt_filename)
         super(FileConsumer, self)._on_complete()
 
+    def _create_files(self, filename):
+        self._filename = filename
+        mkdir_p(os.path.dirname(filename))
```

Comment:
This is where the dirfd stuff I was talking about could be added: when creating the leaf directory, keep an FD to it, and then use the Python equivalent of `openat()` to open the files beneath it. This removes race conditions in resolving the directory path to an inode when creating each of the files. (Note that I haven’t actually thought about the code past this musing.)